### PR TITLE
IGNITE-14591: Add javadoc validation to checkstyle plugin.

### PR DESCRIPTION
**Pull request by AMashenkov**

### CODE REVIEW

File: check-rules/checkstyle-javadoc-additional-rules.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            See: https://checkstyle.org/config_javadoc.html#JavadocStyle
+        -->
+        <module name="JavadocStyle">
+            <property name="scope" value="public"/>
+            <property name="checkEmptyJavadoc" value="true"/>
+        </module>
+
+        <module name="JavadocStyle">
+            <property name="scope" value="private"/>
+            <property name="excludeScope" value="protected"/>
```
Comment (ygerzhedovich) [inline check-rules/checkstyle-javadoc-additional-rules.xml:47]: why we should have exclude for protected scope here?

File: check-rules/checkstyle-javadoc-additional-rules.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        <!--
+            Checks the Javadoc of a method or constructor.
+            See: https://checkstyle.org/config_javadoc.html#JavadocMethod
+        -->
+        <module name="JavadocMethod">
+            <property name="scope" value="protected"/>
+            <property name="validateThrows" value="true"/>
+        </module>
+
+        <module name="JavadocMethod">
```
Comment (ygerzhedovich) [inline check-rules/checkstyle-javadoc-additional-rules.xml:81]: Could you please clarify, the second section will override the first section?

Comment (AMashenkov) [reply check-rules/checkstyle-javadoc-additional-rules.xml:81]: No, there are different rules.

File: parent/pom.xml [hunk 3]
```diff
@@ -596,6 +596,7 @@
         <!--
             Profile to exclude running surefire (unit) tests but do not prevent running integration ones
         -->
+        <!--suppress MavenModelInspection -->
```
Comment (vveider) [inline parent/pom.xml:599]: What does explain this comment?

**Review by vveider — CHANGES_REQUESTED** [review]

**Pull request merged**